core/src/main/java/hudson/model/Queue.java

```diff
@@ -219,7 +220,7 @@
      * to assign a work. Once a work is assigned, the executor actually gets
      * started to carry out the task in question.
      */
-    public class JobOffer extends MappingWorksheet.ExecutorSlot {
+    public static class JobOffer extends MappingWorksheet.ExecutorSlot {
```

Comment:
Indeed I am not really sure why this was ever `public`—it only seems to be used internally in `Queue.maintain`, plus a mock in `LoadPredictorTest`.